src/analyze/analyze-security.c

```diff
@@ -144,12 +149,20 @@ static void security_info_free(struct security_info *i) {
         free(i->device_policy);
 
         strv_free(i->supplementary_groups);
-        strv_free(i->system_call_architectures);
+        set_free(i->system_call_architectures);
+
+        hashmap_free(i->system_call_filter);
+}
 
-        set_free(i->system_call_filter);
+static SecurityInfo *security_info_free_free(SecurityInfo *info) {
```

Comment:
i am not a fan of having two distinct destructors for this structure. Yes, previously the structures was allocated on the stack, but if you go for heap now, then go heap all the way, and always allocate this on the heap. i.e. please drop security_info_free_free() and instead just free the whole struct in regular security_info_free() and make sure to always allocate the structure on the heap.

It's just very confusing if there are wo different ways you can allocate an object like this, that requires careful matchup with destructors. There might be places where that's worth it (inner loops of some kind), but this totally isn't one of them, hence simplify: define one way to allocate and object and one way to free it.